src/Response/DTO/ParcelShop.php

```diff
@@ -4,178 +4,61 @@
-use Answear\GlsBundle\Util\BooleanTransformer;
+use Answear\GlsBundle\Enum\CountryCodeEnum;
+use Answear\GlsBundle\Enum\FeatureEnum;
+use Answear\GlsBundle\Enum\ParcelShopTypeEnum;
 
 class ParcelShop
 {
-    private string $shopId;
-    private string $name;
-    private CountryCode $countryCode;
-    private bool $codHandler;
-    private bool $payByBankCard;
-    private bool $dropOffPoint;
-    private string $owner;
-    private bool $parcelLocker;
-    private ?string $vendorUrl;
-    private ?string $pickupTime;
-    private ?string $info;
-    private Address $address;
     /**
-     * @var Openings[]
+     * @param Openings[] $openings
      */
-    private array $openings = [];
-    private ?Coordinates $coordinates;
-    private ?string $holidayStarts;
-    private ?string $holidayEnds;
-
     public function __construct(
-        string $shopId,
-        string $name,
-        CountryCode $countryCode,
-        bool $isCodHandler,
-        bool $payByBankCard,
-        bool $dropOffPoint,
-        string $owner,
-        bool $isParcelLocker,
-        ?string $vendorUrl,
-        ?string $pickupTime,
-        ?string $info,
-        Address $address,
-        array $openings,
-        ?Coordinates $coordinates,
-        ?string $holidayStarts,
-        ?string $holidayEnds
+        public string $shopId,
+        public string $name,
+        public CountryCodeEnum $countryCode,
+        public bool $isCodHandler,
+        public bool $payByBankCard,
+        public bool $isParcelLocker,
+        public ?string $pickupTime,
+        public ?string $info,
+        public Address $address,
+        public array $openings,
+        public Coordinates $coordinates,
+        public array $features,
     ) {
-        $this->shopId = $shopId;
-        $this->name = $name;
-        $this->countryCode = $countryCode;
-        $this->codHandler = $isCodHandler;
-        $this->payByBankCard = $payByBankCard;
-        $this->dropOffPoint = $dropOffPoint;
-        $this->owner = $owner;
-        $this->parcelLocker = $isParcelLocker;
-        $this->vendorUrl = $vendorUrl;
-        $this->pickupTime = $pickupTime;
-        $this->info = $info;
-        $this->address = $address;
-        $this->openings = $openings;
-        $this->coordinates = $coordinates;
-        $this->holidayStarts = $holidayStarts;
-        $this->holidayEnds = $holidayEnds;
     }
 
-    /**
-     * @param Openings[] $openings
-     */
-    public static function fromRawParcelShop(RawParcelShop $rawParcelShop, array $openings): self
+    public static function fromRawParcelShop(RawParcelShop $rawParcelShop): self
     {
-        $coordinates = null;
-
-        if (null !== $rawParcelShop->geolat && null !== $rawParcelShop->geolng) {
-            $coordinates = new Coordinates((float) $rawParcelShop->geolat, (float) $rawParcelShop->geolng);
-        }
+        $isCodHandler = in_array(FeatureEnum::AcceptsCash->value, $rawParcelShop->features, true);
```

Comment:
These variables are not reused, in my opinion they do not make order here at all just more confusion. I propose to remove the variables.

Reply:
Not a problem, I shortened the list of variables.